owrrm/src/main/java/com/facebook/openwifi/rrm/rca/RCAUtils.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.openwifi.rrm.rca;
+
+import com.facebook.openwifi.cloudsdk.UCentralClient;
+import com.facebook.openwifi.cloudsdk.UCentralUtils;
+import com.facebook.openwifi.cloudsdk.models.gw.CommandInfo;
+
+/**
+ * Utilities for root cause analysis.
+ */
+public class RCAUtils {
+	/** Ping result, only containing a data summary (not individual pings). */
+	public static class PingResult {
+		public double min;
+		public double avg;
+		public double max;
+		public double mdev;
+		// TODO other stats? (ex. tx/rx packets, % packet loss)
+
+		@Override
+		public String toString() {
+			return String.format("%.3f/%.3f/%.3f/%.3f ms", min, avg, max, mdev);
+		}
+	}
+
+	/**
+	 * Parse raw ping output, returning null upon error.
+	 *
+	 * This only supports the busybox ping format.
+	 */
+	private static PingResult parsePingOutput(String output) {
+		// Only parse summary line (should be last line in output).
+		// Code below is optimized for minimal string operations.
+		//
+		// Examples of supported formats:
+		//   round-trip min/avg/max = 4.126/42.470/84.081 ms
+		//   rtt min/avg/max/mdev = 16.853/20.114/23.375/3.261 ms
```

Comment:
can we use regex with matching groups instead? 

```
min\/avg\/max(?:\/mdev)? = (\d+\.\d+)\/(\d+\.\d+)\/(\d+\.\d+)(?:\/(\d+\.\d+))?
```
then we can use capture groups to refer to the fields directly

Reply:
see comment above, this is optimizing for string parsing time since regex is several orders of magnitude slower than what I'm doing (and the manual parsing isn't that painful)